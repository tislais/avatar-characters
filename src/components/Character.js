import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, allies, enemies, photoUrl, id }) => (

  <figure className="">
      <img src={photoUrl} alt={name}
        className="
          h-[300px] sm:h-[200px] 
          w-full 
          object-cover
        " />
      <figcaption>
        <p className="uppercase text-lg font-bold tracking-wider">
          {name}
        </p>

        <p>Allies: {allies.map((ally) => (
          <span key={id}>{ally}</span>
        ))}
        </p>

        <p>Enemies: {enemies.map((enemy) => (
          <span>{enemy}</span>
        ))}
        </p>
      </figcaption>
    </figure>

);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  allies: PropTypes.array,
  enemies: PropTypes.array,
  photoUrl: PropTypes.string.isRequired
};

export default Character;