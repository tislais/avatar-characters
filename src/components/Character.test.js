import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders Character', () => {
    const { asFragment } = render(<Character
      name="46th Earth King"
      allies={["Royal Earthbender Guards"]}
      enemies={["Chin"]}
      photo="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441" />);
    expect(asFragment()).toMatchSnapshot();
  });
});