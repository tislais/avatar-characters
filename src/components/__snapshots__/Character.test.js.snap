// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure
    class=""
  >
    <img
      alt="46th Earth King"
      class="
 h-[300px] sm:h-[200px] 
 w-full 
 object-cover
 "
      src="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441"
    />
    <figcaption>
      <p
        class="uppercase text-lg font-bold tracking-wider"
      >
        46th Earth King
      </p>
      <p>
        Allies: 
        <span>
          Royal Earthbender Guards
        </span>
      </p>
      <p>
        Enemies: 
        <span>
          Chin
        </span>
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
